exercises/05.actions/05.solution.history-revalidation/server/app.js

```diff
@@ -1,66 +1,72 @@
-import bodyParser from 'body-parser'
-import busboy from 'busboy'
+import { readFile } from 'fs/promises'
+import { serve } from '@hono/node-server'
+import { serveStatic } from '@hono/node-server/serve-static'
+import { RESPONSE_ALREADY_SENT } from '@hono/node-server/utils/response'
 import closeWithGrace from 'close-with-grace'
-import compress from 'compression'
-import express from 'express'
+import { Hono } from 'hono'
+import { trimTrailingSlash } from 'hono/trailing-slash'
 import { createElement as h } from 'react'
 import {
 	renderToPipeableStream,
-	decodeReplyFromBusboy,
+	decodeReply,
 } from 'react-server-dom-esm/server'
 import { App } from '../src/app.js'
 import { shipDataStorage } from './async-storage.js'
 
 const PORT = process.env.PORT || 3000
 
-const app = express()
-app.use(compress())
-// this is here so the workshop app knows when the server has started
-app.head('/', (req, res) => res.status(200).end())
+const app = new Hono({ strict: true })
 
-app.use(express.static('public', { index: false }))
-app.use('/js/src', express.static('src'))
+app.use(trimTrailingSlash())
 
-// This just cleans up the URL if the search ever gets cleared... Not important
-// for RSCs... Just ... I just can't help myself. I like URLs clean.
-app.use((req, res, next) => {
-	if (req.query.search === '') {
-		const searchParams = new URLSearchParams(req.search)
-		searchParams.delete('search')
-		const location = [req.path, searchParams.toString()]
-			.filter(Boolean)
-			.join('?')
-		return res.redirect(302, location)
-	} else {
-		next()
-	}
-})
+app.use(
+	'/js/src/*',
+	serveStatic({
+		root: './src',
+		onNotFound(path, c) {
+			c.text('Not found', 404)
+		},
+		rewriteRequestPath: path => path.replace('/js/src', ''),
+	}),
+)
+
+app.use(
+	serveStatic({
+		root: './public',
+	}),
+	async (c, next) => {
+		await next()
+	},
+)
 
 const moduleBasePath = new URL('../src', import.meta.url).href
 
-async function renderApp(res, returnValue) {
+async function renderApp(c, returnValue) {
 	try {
-		const shipId = res.req.params.shipId || null
-		const search = res.req.query.search || ''
+		const { outgoing } = c.env
+		const url = new URL(c.req.url)
+		const shipId = c.req.param('shipId') || null
+		const search = url.searchParams.get('search') || ''
 		const data = { shipId, search }
 		shipDataStorage.run(data, () => {
 			const root = h(App)
 			const payload = { root, returnValue }
 			const { pipe } = renderToPipeableStream(payload, moduleBasePath)
-			pipe(res)
+			pipe(outgoing)
 		})
+		return RESPONSE_ALREADY_SENT
 	} catch (error) {
 		console.error(error)
-		res.status(500).json({ error: error.message })
+		return c.json({ error: error.message }, 500)
 	}
```

Comment:
Because of the error handler below I think we can remove this and ditch the try/catch.